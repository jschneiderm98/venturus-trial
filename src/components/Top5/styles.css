.Top5 {
  margin-left: 100px;
  box-shadow: 0px 0px 1px 1px #f3ecf2;
  background-color: #ffffff;
  width: 600px;
  height: 350px;
  border-radius: 10px;
}

.ComponentHeader {
  height: 80px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  font-size: 24px;
  font-weight: 700;
  border-bottom: 1px solid #f8f1f7;
  color: #64008b;
  text-indent: 30px;
}

.ComponentBody {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.Highest,
.Lowest {
  margin-right: 10px;
  margin-top: 15px;
  font-weight: 500;
  text-align: start;
  color: #333333;
}

.ItemList {
  height: 200px;
  width: 240px;
  background-color: #f8f1f7;
  border-radius: 10px;
  display: flex;
  justify-content: space-evenly;
  margin-top: 10px;
}

.Item {
  width: 230px;
  height: 34px;
  border-radius: 5px;
  background-color: #ffffff;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.Item:first-child {
  margin-top: 2px;
}

.Item>td:first-child {
  padding-left: 10px;
  font-weight: 400;
  font-size: 12px;
}

.Item>td:nth-child(2) {
  padding-right: 10px;
  font-weight: 700;
  font-size: 12px;
}