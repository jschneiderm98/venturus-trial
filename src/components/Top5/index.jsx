import React from 'react';
import './styles.css';

export default function Top5() {
  const HighestTeams = [
    { name: 'Inter Milan', avg: 31.9 },
    { name: 'APOEL Nicosia', avg: 31.7 },
    { name: 'AC Milan', avg: 31.6 },
    { name: 'Besiktas JK', avg: 31.4 },
    { name: 'Olympiacos Piraeus', avg: 31.3 },
  ];
  const LowestTeams = [
    { name: 'Zalgiris Vilnius', avg: 21.1 },
    { name: 'Arsenal FC', avg: 21.6 },
    { name: 'Ajax Amsterdam', avg: 22.0 },
    { name: 'FC Nantes', avg: 22.1 },
    { name: 'CSKA Moskow', avg: 22.5 },
  ];
  return (
    <div className="Top5">
      <div className="ComponentHeader">
        Top5
      </div>
      <div className="ComponentBody">
        <div className="Highest">
          <span>Highest avg age</span>
          <div className="ItemList">
            <table>
              {HighestTeams && HighestTeams.map((team) => (
                <tr>
                  <div className="Item">
                    <td>{team.name}</td>
                    <td>{team.avg}</td>
                  </div>
                </tr>
              ))}
            </table>
          </div>
        </div>
        <div className="Lowest">
          <span>Lowest avg age</span>
          <div className="ItemList">
            <table>
              {LowestTeams && LowestTeams.map((team) => (
                <tr>
                  <div className="Item">
                    <td>{team.name}</td>
                    <td>{team.avg}</td>
                  </div>
                </tr>
              ))}
            </table>
          </div>
        </div>
      </div>
    </div>
  );
}
