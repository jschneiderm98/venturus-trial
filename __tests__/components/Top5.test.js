import React from 'react';
import { render } from '@testing-library/react';
import Top5 from '../../src/components/Top5';

describe('Testing Top5 component', () => {
  test('Should render', () => {
    const tree = render(<Top5 />);

    expect(tree).toMatchSnapshot();
  });
});
