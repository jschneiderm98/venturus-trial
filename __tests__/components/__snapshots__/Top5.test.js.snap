// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Top5 component Should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Top5"
      >
        <div
          class="ComponentHeader"
        >
          Top5
        </div>
        <div
          class="ComponentBody"
        >
          <div
            class="Highest"
          >
            <span>
              Highest avg age
            </span>
            <div
              class="ItemList"
            >
              <table>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
              </table>
            </div>
          </div>
          <div
            class="Lowest"
          >
            <span>
              Lowest avg age
            </span>
            <div
              class="ItemList"
            >
              <table>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
                <tr>
                  <div
                    class="Item"
                  >
                    <td>
                      Inter Milan
                    </td>
                    <td>
                      31.9
                    </td>
                  </div>
                </tr>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Top5"
    >
      <div
        class="ComponentHeader"
      >
        Top5
      </div>
      <div
        class="ComponentBody"
      >
        <div
          class="Highest"
        >
          <span>
            Highest avg age
          </span>
          <div
            class="ItemList"
          >
            <table>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
            </table>
          </div>
        </div>
        <div
          class="Lowest"
        >
          <span>
            Lowest avg age
          </span>
          <div
            class="ItemList"
          >
            <table>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
              <tr>
                <div
                  class="Item"
                >
                  <td>
                    Inter Milan
                  </td>
                  <td>
                    31.9
                  </td>
                </div>
              </tr>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
